src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -263,9 +264,9 @@ private GenericInfoBucketData createGenericInfoBucketData(BucketEntity bucket) {
         return GenericInfoBucketData.builder().bucketName(bucket.getBucketName()).group(bucket.getOwner()).build();
     }
 
-    public List<CatalogObjectMetadata> listCatalogObjects(String bucketName) {
-        findBucketByNameAndCheck(bucketName);
-        List<CatalogObjectRevisionEntity> result = catalogObjectRevisionRepository.findDefaultCatalogObjectsInBucket(bucketName);
+    public List<CatalogObjectMetadata> listCatalogObjects(List<String> bucketNames) {
+        bucketNames.forEach(bucketName -> findBucketByNameAndCheck(bucketName));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)